src/kudu/consensus/raft_consensus.cc

```diff
@@ -1031,7 +1031,7 @@ Status RaftConsensus::AddPendingOperationUnlocked(const scoped_refptr<ConsensusR
 
     const RaftConfigPB& new_config = change_record->new_config();
 
-    if (!new_config.unsafe_config_change()) {
+    if (adjust_voter_distribution_ && !new_config.unsafe_config_change()) {
```

Comment:
Since voter dist adjustment is enabled/disabled by the python process there is no guarantee that some automation that is trying to change the config (e.g. add/remove) does not race with UCC. This means a pending op can be appended while UCC is still working on fixing the ring by disabling voter dist adj. While we make sure such scenarios are not common by taking a dblock on the ring, I think it is much better to force the UCC thru instead of erroring out. You can also see that kudu's UCC also skips this check.

Reply:
I have some concerns on skipping this change, because pending config change is allowed after UCC is live...
but I would like to unblock this for now. We can revisit this through production experience.